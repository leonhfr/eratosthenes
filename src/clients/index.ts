// Packages.
import * as AWS from 'aws-sdk';

// Instanciation.
export const DynamoDB = new AWS.DynamoDB.DocumentClient();
export const Lambda = new AWS.Lambda();
export const S3 = new AWS.S3();
export const Sqs = new AWS.SQS();
