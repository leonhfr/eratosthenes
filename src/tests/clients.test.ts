import * as clients from '../clients';

// Code.
describe('clients', () => {
  it('should expose the expected entries', () => {
    expect(clients).toMatchSnapshot();
  });
});
