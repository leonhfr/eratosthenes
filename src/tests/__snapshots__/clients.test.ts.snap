// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`clients should expose the expected entries 1`] = `
Object {
  "DynamoDB": DocumentClient {
    "attrValue": "S8",
    "options": Object {
      "attrValue": "S8",
    },
    "service": Object {
      "CALL_EVENTS_BUBBLE": [Function],
      "MONITOR_EVENTS_BUBBLE": [Function],
      "_clientId": 1,
      "_events": Object {
        "apiCall": Array [
          [Function],
        ],
        "apiCallAttempt": Array [
          [Function],
        ],
      },
      "config": Config {
        "apiVersion": null,
        "apiVersions": Object {},
        "clientSideMonitoring": false,
        "computeChecksums": true,
        "convertResponseTypes": true,
        "correctClockSkew": false,
        "credentialProvider": CredentialProviderChain {
          "providers": Array [
            [Function],
            [Function],
            [Function],
            [Function],
          ],
          "resolveCallbacks": Array [],
        },
        "credentials": null,
        "customUserAgent": null,
        "dynamoDbCrc32": true,
        "endpoint": "dynamodb.undefined.amazonaws.com",
        "endpointCacheSize": 1000,
        "endpointDiscoveryEnabled": false,
        "hostPrefixEnabled": true,
        "httpOptions": Object {
          "timeout": 120000,
        },
        "logger": null,
        "maxRedirects": 10,
        "maxRetries": undefined,
        "paramValidation": true,
        "region": undefined,
        "retryDelayOptions": Object {},
        "s3BucketEndpoint": false,
        "s3DisableBodySigning": true,
        "s3ForcePathStyle": false,
        "signatureCache": true,
        "signatureVersion": "v4",
        "sslEnabled": true,
        "systemClockOffset": 0,
        "useAccelerateEndpoint": false,
      },
      "endpoint": Endpoint {
        "host": "dynamodb.undefined.amazonaws.com",
        "hostname": "dynamodb.undefined.amazonaws.com",
        "href": "https://dynamodb.undefined.amazonaws.com/",
        "path": "/",
        "pathname": "/",
        "port": 443,
        "protocol": "https:",
      },
      "isGlobalEndpoint": false,
    },
  },
  "Lambda": Object {
    "CALL_EVENTS_BUBBLE": [Function],
    "MONITOR_EVENTS_BUBBLE": [Function],
    "_clientId": 2,
    "_events": Object {
      "apiCall": Array [
        [Function],
      ],
      "apiCallAttempt": Array [
        [Function],
      ],
    },
    "config": Config {
      "apiVersion": null,
      "apiVersions": Object {},
      "clientSideMonitoring": false,
      "computeChecksums": true,
      "convertResponseTypes": true,
      "correctClockSkew": false,
      "credentialProvider": CredentialProviderChain {
        "providers": Array [
          [Function],
          [Function],
          [Function],
          [Function],
        ],
        "resolveCallbacks": Array [],
      },
      "credentials": null,
      "customUserAgent": null,
      "dynamoDbCrc32": true,
      "endpoint": "lambda.undefined.amazonaws.com",
      "endpointCacheSize": 1000,
      "endpointDiscoveryEnabled": false,
      "hostPrefixEnabled": true,
      "httpOptions": Object {
        "timeout": 120000,
      },
      "logger": null,
      "maxRedirects": 10,
      "maxRetries": undefined,
      "paramValidation": true,
      "region": undefined,
      "retryDelayOptions": Object {},
      "s3BucketEndpoint": false,
      "s3DisableBodySigning": true,
      "s3ForcePathStyle": false,
      "signatureCache": true,
      "signatureVersion": "v4",
      "sslEnabled": true,
      "systemClockOffset": 0,
      "useAccelerateEndpoint": false,
    },
    "endpoint": Endpoint {
      "host": "lambda.undefined.amazonaws.com",
      "hostname": "lambda.undefined.amazonaws.com",
      "href": "https://lambda.undefined.amazonaws.com/",
      "path": "/",
      "pathname": "/",
      "port": 443,
      "protocol": "https:",
    },
    "isGlobalEndpoint": false,
  },
  "S3": Object {
    "CALL_EVENTS_BUBBLE": [Function],
    "MONITOR_EVENTS_BUBBLE": [Function],
    "_clientId": 3,
    "_events": Object {
      "apiCall": Array [
        [Function],
      ],
      "apiCallAttempt": Array [
        [Function],
      ],
    },
    "config": Config {
      "apiVersion": null,
      "apiVersions": Object {},
      "clientSideMonitoring": false,
      "computeChecksums": true,
      "convertResponseTypes": true,
      "correctClockSkew": false,
      "credentialProvider": CredentialProviderChain {
        "providers": Array [
          [Function],
          [Function],
          [Function],
          [Function],
        ],
        "resolveCallbacks": Array [],
      },
      "credentials": null,
      "customUserAgent": null,
      "dynamoDbCrc32": true,
      "endpoint": "s3.amazonaws.com",
      "endpointCacheSize": 1000,
      "endpointDiscoveryEnabled": false,
      "hostPrefixEnabled": true,
      "httpOptions": Object {
        "timeout": 120000,
      },
      "logger": null,
      "maxRedirects": 10,
      "maxRetries": undefined,
      "paramValidation": true,
      "region": "us-east-1",
      "retryDelayOptions": Object {},
      "s3BucketEndpoint": false,
      "s3DisableBodySigning": true,
      "s3ForcePathStyle": false,
      "signatureCache": true,
      "signatureVersion": "s3",
      "sslEnabled": true,
      "systemClockOffset": 0,
      "useAccelerateEndpoint": false,
    },
    "endpoint": Endpoint {
      "host": "s3.amazonaws.com",
      "hostname": "s3.amazonaws.com",
      "href": "https://s3.amazonaws.com/",
      "path": "/",
      "pathname": "/",
      "port": 443,
      "protocol": "https:",
    },
    "isGlobalEndpoint": false,
  },
  "Sqs": Object {
    "CALL_EVENTS_BUBBLE": [Function],
    "MONITOR_EVENTS_BUBBLE": [Function],
    "_clientId": 4,
    "_events": Object {
      "apiCall": Array [
        [Function],
      ],
      "apiCallAttempt": Array [
        [Function],
      ],
    },
    "config": Config {
      "apiVersion": null,
      "apiVersions": Object {},
      "clientSideMonitoring": false,
      "computeChecksums": true,
      "convertResponseTypes": true,
      "correctClockSkew": false,
      "credentialProvider": CredentialProviderChain {
        "providers": Array [
          [Function],
          [Function],
          [Function],
          [Function],
        ],
        "resolveCallbacks": Array [],
      },
      "credentials": null,
      "customUserAgent": null,
      "dynamoDbCrc32": true,
      "endpoint": "sqs.undefined.amazonaws.com",
      "endpointCacheSize": 1000,
      "endpointDiscoveryEnabled": false,
      "hostPrefixEnabled": true,
      "httpOptions": Object {
        "timeout": 120000,
      },
      "logger": null,
      "maxRedirects": 10,
      "maxRetries": undefined,
      "paramValidation": true,
      "region": undefined,
      "retryDelayOptions": Object {},
      "s3BucketEndpoint": false,
      "s3DisableBodySigning": true,
      "s3ForcePathStyle": false,
      "signatureCache": true,
      "signatureVersion": "v4",
      "sslEnabled": true,
      "systemClockOffset": 0,
      "useAccelerateEndpoint": false,
    },
    "endpoint": Endpoint {
      "host": "sqs.undefined.amazonaws.com",
      "hostname": "sqs.undefined.amazonaws.com",
      "href": "https://sqs.undefined.amazonaws.com/",
      "path": "/",
      "pathname": "/",
      "port": 443,
      "protocol": "https:",
    },
    "isGlobalEndpoint": false,
  },
}
`;
